src/Orleans.CodeGenerator/CopierGenerator.cs

```diff
@@ -209,14 +209,17 @@ static ExpressionSyntax GetService(TypeSyntax type)
                 fields.Add(new BaseCopierFieldDescription(libraryTypes.BaseCopier_1.ToTypeSyntax(serializableTypeDescription.BaseTypeSyntax), BaseTypeCopierFieldName));
             }
 
-            if (serializableTypeDescription.UseActivator)
+            if (!serializableTypeDescription.IsUnsealedImmutable)
```

Comment:
What's the rationale behind this guard? I'm not quite understanding the `sealed` part, specifically.
I'm interpreting the code as: *If the type is either sealed or mutable, allow it to have an activator and field copiers.*
Am I confused in thinking that a `sealed` type may still need an activator and field copiers?

Reply:
If the type is sealed and immutable, then no copier is generated at all. If it's unsealed but immutable, then a copier is generated to support derived types (but no activator nor field copiers are needed).
In this particular location the condition could be written just `!IsImmutable`, but I re-used `!IsUnsealedImmutable` here because it works just as well (and exists).

Reply:
Ok, LGTM. I pushed a test, rebased it, and added `IsImmutable` to clear this logic up (seeing as `sealed` is not logically part of this condition).